macros/playground/src/lib.rs

```diff
@@ -231,74 +231,43 @@ mod tests {
     #[job(
         id = 0,
         params(value),
-
         event_listener(
-            listener = PeriodicEventListener<1500, WebPoller, serde_json::Value, Arc<AtomicUsize>>,
+            listener = PeriodicEventListener<1500, WebPollerTest, serde_json::Value, Arc<AtomicUsize>>,
             pre_processor = pre_process,
             post_processor = post_process,
         ),
     )]
     // Maps a boolean value obtained from pre-processing to a u8 value
-    pub async fn web_poller(value: bool, count: Arc<AtomicUsize>) -> Result<u8, Infallible> {
+    pub async fn web_poller_test(value: bool, count: Arc<AtomicUsize>) -> Result<u8, Infallible> {
```

Comment:
Can't we test this in the blueprint examples repo? It's getting to be a bit of a nested mess IMO. PeriodicListener with WebPoller and then another PeriodicListener with a WebPollerTest containing a WebPoller.

I would suggest getting CI to test the example blueprint instead? Would that accomplish your goal here? It would also be good to have a PeriodicListener example as a simple cron-job no inner listener too, no complexity, just "here's how you trigger your job every 2 seconds".

Reply:
We currently don't have a clean way of testing blueprints unless we inject test-related code into them (like I did for the `web_poller_test`). While we could test the one inside `examples`, we would have to add more code to it, confusing the user by mixing prod code with test code. At least with `web_poller_test`, I just need to import the already-existing code inside `examples`, then add a thin wrapper over it where I am then able to inject the test-related code.